### PR TITLE
AKS Baseline regulated - New Azure Defender for containers feature introduced.

### PR DESCRIPTION
**Pull request by magrande**

### CODE REVIEW

File: docs/deploy/01-prerequisites.md [hunk 1]
```diff
@@ -35,9 +35,11 @@ Throughout this walkthrough, take note of the following symbol.
 
    ```bash
    az feature register --namespace "Microsoft.ContainerService" -n "EnablePodIdentityPreview"
+   az feature register --namespace "Microsoft.ContainerService" -n "AKS-AzureDefender"
```
Comment (ckittel) [inline docs/deploy/01-prerequisites.md:38]: There is a missing unordered list item above this change.  Please follow the pattern shown for Pod Identity.  But just to verify, is this even needed anymore now that it's GA?  I don't see this in the getting started instructions on docs.microsoft.com anymore.

File: docs/deploy/04-subscription.md [hunk 2]
```diff
@@ -27,7 +27,8 @@ To help govern our resources, there are policies we apply over the scope of thes
 
 | Policy Name                    | Scope                           | Purpose                                                                                           |
 |--------------------------------|---------------------------------|---------------------------------------------------------------------------------------------------|
-| Enable Microsoft Defender Standard | Subscription                | Ensures that Microsoft Defender for Kubernetes, Container Service, and Key Vault are always enabled. |
+| Enable Microsoft Defender Standard | Subscription                | Ensures that Microsoft Defender for Containers, and Key Vault are always enabled. |
+| Enable Microsoft Defender for Containers in the cluster | rg-bu0001a0005                | Ensures that Microsoft Defender for Containers is enabled in the cluster. |
```
Comment (ckittel) [inline docs/deploy/04-subscription.md:31]: Hum, is that what the policy is?  I thought it was a "No AKS clusters without Microsoft Defender for Containers" policy?  You have it here implied as a DINE policy.  Please move "No AKS clusters without..." policy down further in the table to sit along side the others in this table.

File: docs/deploy/04-subscription.md [hunk 3]
```diff
@@ -27,7 +27,8 @@ To help govern our resources, there are policies we apply over the scope of thes
 
 | Policy Name                    | Scope                           | Purpose                                                                                           |
 |--------------------------------|---------------------------------|---------------------------------------------------------------------------------------------------|
-| Enable Microsoft Defender Standard | Subscription                | Ensures that Microsoft Defender for Kubernetes, Container Service, and Key Vault are always enabled. |
+| Enable Microsoft Defender Standard | Subscription                | Ensures that Microsoft Defender for Containers, and Key Vault are always enabled. |
```
Comment (ckittel) [inline docs/deploy/04-subscription.md:30]:
```suggestion
| Enable Microsoft Defender Standard | Subscription                | Ensures that Microsoft Defender for Containers, DNS, Key Vault, and Resource Manager are always enabled. |
```

File: docs/deploy/04-subscription.md [hunk 4]
```diff
@@ -55,8 +56,7 @@ Also, depending on your workload subscription scope, some of the policies applie
 
 As mentioned in the Azure Policy section above, we enable the following Microsoft Defender for Cloud's services.
 
-* [Microsoft Defender for Kubernetes](https://docs.microsoft.com/azure/security-center/defender-for-kubernetes-introduction)
-* [Microsoft Defender for Container Registries](https://docs.microsoft.com/azure/security-center/defender-for-container-registries-introduction)
+* [Microsoft Defender for Containers](https://docs.microsoft.com/azure/defender-for-cloud/defender-for-containers-introduction?tabs=defender-for-container-arch-aks)
 * [Microsoft Defender for Key Vault](https://docs.microsoft.com/azure/security-center/defender-for-key-vault-introduction)
```
Comment (ckittel) [inline docs/deploy/04-subscription.md:60]: Can you update these three existing links to their new home?  E.g. this one would be. https://docs.microsoft.com/azure/defender-for-cloud/defender-for-key-vault-introduction

File: docs/deploy/10-pre-bootstrap.md [hunk 5]
```diff
@@ -16,7 +16,7 @@ You'll end up with the following images imported into your ACR instance, after h
...
... images to repository names that starts with `quarantine/`. We'll then show you Microsoft Defender for containers scan of those images, and then you'll import those same images directly from `quarantine/` to `live/` repositories (retaining their sha256 digest). We've restricted our cluster to only allow pulling from `live/` repositories and we've built an alert if an image was imported to `live/` from a source other than `quarantine/`. This isn't a preventative security control; _this won't block a direct import_ request or _validate_ that the image actually passed quarantine checks. There are other solutions you can use for this pattern that are more exhaustive. [Aquasec](https://go.microsoft.com/fwlink/?linkid=2002601&clcid=0x409) and [Twistlock](https://go.microsoft.com/fwlink/?linkid=2002600&clcid=0x409) both offer integrated solutions specifically for Azure Container Registry scanning and compliance management. Azure Container Registry has an [integrated quarantine feature](https://docs.microsoft.com/azure/container-registry/container-registry-faq#how-do-i-enable-automatic-image-quarantine-for-a-registry) as well that could be considered, however it is in preview at this time.
```
Comment (ckittel) [inline docs/deploy/10-pre-bootstrap.md:19]:
```suggestion
Quarantining first- and third-party images is a recommended security practice. This allows you to get your images onto a dedicated container registry and subject them to any sort of security/compliance scrutiny you wish to apply. Once validated, they can then be promoted to being available to your cluster. There are many variations on this pattern, with different tradeoffs for each. For simplicity in this walkthrough we are simply going to import our images to repository names that starts with `quarantine/`. We'll then show you Microsoft Defender for Containers' scan of those images, and then you'll import those same images directly from `quarantine/` to `live/` repositories (retaining their sha256 digest). We've restricted our cluster to only allow pulling from `live/` repositories and we've built an alert if an image was imported to `live/` from a source other than `quarantine/`. This isn't a preventative security control; _this won't block a direct import_ request or _validate_ that the image actually passed quarantine checks. There are other solutions you can use for this pattern that are more exhaustive. [Aquasec](https://go.microsoft.com/fwlink/?linkid=2002601&clcid=0x409) and [Twistlock](https://go.microsoft.com/fwlink/?linkid=2002600&clcid=0x409) both offer integrated solutions specifically for Azure Container Registry scanning and compliance management. Azure Container Registry has an [integrated quarantine feature](https://docs.microsoft.com/azure/container-registry/container-registry-faq#how-do-i-enable-automatic-image-quarantine-for-a-registry) as well that could be considered, however it is in preview at this time.
```

File: docs/deploy/11-gitops.md [hunk 6]
```diff
@@ -219,7 +219,7 @@ Common features offered in ISV solutions like these:
...
...he default rules as they were defined when its manifests were generated. This is also being installed for illustrative purposes, and you're encouraged to evaluate if a solution like Falco is relevant to you. If so, in your final implementation, review and tune its deployment to fit your needs (E.g. add custom rules like [CVE detection](https://artifacthub.io/packages/search?ts_query_web=cve&org=falco), [sudo usage](https://artifacthub.io/packages/falco/security-hub/admin-activities), [basic FIM](https://artifacthub.io/packages/falco/security-hub/file-integrity-monitoring), [SSH Connection monitoring](https://artifacthub.io/packages/falco/security-hub/ssh-connections), and [NGINX containment](https://artifacthub.io/packages/falco/security-hub/nginx)). This tooling, _as most security tooling will be_, is **highly-privileged within your cluster**. Usually running as DaemonSets with access to the underlying node in a manor that is well beyond any typical workload in your cluster. Remember to consider the runtime compute and networking requirements of your security tooling when sizing your cluster, as these can often be overlooked when initial cluster sizing conversations are happening.
```
Comment (ckittel) [inline docs/deploy/11-gitops.md:222]:
```suggestion
This reference implementation includes Microsoft Defender for Containers as a solution for helping you secure your containers and runtime environment. Microsoft Defender for Containers, provides [security alerts](https://docs.microsoft.com/azure/defender-for-cloud/alerts-reference#alerts-k8scluster) on the cluster level and on the underlying cluster nodes by monitoring both the API server activity and the behavior of the workloads themselves.   

You can also consider installing third party solutions to supplement Microsoft Defender for Containers as a defense-in-depth strategy. As an example of that, in this reference implementation you also install a very basic deployment of [Falco](https://falco.org/). It is not configured for alerts, nor tuned to any specific needs. It uses the default rules as they were defined when its manifests were generated. This is installed for illustrative purposes, and you're encouraged to evaluate if a third-party solution like Falco is relevant to your situation. If so, in your final implementation, review and tune its deployment to fit your needs (E.g. add custom rules like [CVE detection](https://artifacthub.io/packages/search?ts_query_web=cve&org=falco), [sudo usage](https://artifacthub.io/packages/falco/security-hub/admin-activities), [basic FIM](https://artifacthub.io/packages/falco/security-hub/file-integrity-monitoring), [SSH Connection monitoring](https://artifacthub.io/packages/falco/security-hub/ssh-connections), and [NGINX containment](https://artifacthub.io/packages/falco/security-hub/nginx)). This tooling, _as most security tooling will be_, is **highly privileged within your cluster**. Usually running as DaemonSets with access to the underlying node in a manner that is well beyond any typical workload in your cluster. Remember to consider the runtime compute and networking requirements of your security tooling when sizing your cluster, as these can often be overlooked when initial cluster sizing conversations are happening.
```

File: docs/deploy/11-gitops.md [hunk 7]
```diff
@@ -219,7 +219,7 @@ Common features offered in ISV solutions like these:
...
 
 It's worth repeating again, **most customers with regulated workloads are bringing ISV or open source security solutions to their clusters**. Azure Kubernetes Service is a managed Kubernetes platform, it does not imply that you will exclusively be using Microsoft products/solutions to solve your requirements. For the most part, after the deployment of the infrastructure and some out-of-the-box addons (like Azure Policy, Azure Monitor, Azure AD Pod Identity, Open Service Mesh), you're in charge of what you choose to run in your hosted Kubernetes platform. Bring the business and compliance solving solutions you need to the cluster from the [vast and ever-growing Kubernetes and CNCF ecosystem](https://l.cncf.io/?fullscreen=yes).
```
Comment (ckittel) [inline docs/deploy/11-gitops.md:224]:
```suggestion
It's worth noting, some customers with regulated workloads are bringing ISV or open-source security solutions to their clusters in addition to Microsoft Defender for Containers for maximum (and/or redundant) compliance coverage. Azure Kubernetes Service is a managed Kubernetes platform, it does not imply that you will exclusively be using Microsoft products/solutions to solve your requirements. For the most part, after the deployment of the infrastructure and some out-of-the-box addons (like Azure Policy, Azure Monitor, Azure AD Pod Identity, Open Service Mesh), you're in charge of what you choose to run in your hosted Kubernetes platform. Bring the business and compliance solving solutions you need to the cluster from the [vast and ever-growing Kubernetes and CNCF ecosystem](https://l.cncf.io/?fullscreen=yes).
```

File: docs/deploy/13-validation-logs.md [hunk 8]
```diff
@@ -61,9 +61,9 @@ ContainerRegistryRepositoryEvents
 
 Unless you performed some additional steps beyond this walkthrough's instructions, you'll notice that the value for all of the records share the same `Identity` -- that is the Object ID of the AKS cluster's managed identity, specifically kubelet (`<your-cluster-name>-agentpool`).
 
-### Microsoft Defender for container registries
+### Microsoft Defender for containers
```
Comment (ckittel) [inline docs/deploy/13-validation-logs.md:64]:
```suggestion
### Microsoft Defender for Containers
```

File: docs/deploy/13-validation-logs.md [hunk 9]
```diff
@@ -195,7 +195,7 @@ If your subscription has the **Azure Security Benchmark** Azure Policy initiativ
 
 ### Review Security Alerts
 
-Microsoft Defender for kubernetes reviews your cluster's logs and detects behavior that might be undesired. Those alerts surface in the Azure Resource Graph (`securityresources` -> `microsoft.security/locations/alerts`) and also in Microsoft Defender for Cloud in the Azure Portal. Within about 24 hours of your cluster being up and running, you should start to see some alerts show up (Low and Medium).
+Microsoft Defender for Containers reviews your cluster's logs and detects behavior that might be undesired. Those alerts surface in the Azure Resource Graph (`securityresources` -> `microsoft.security/locations/alerts`) and also in Microsoft Defender for Cloud in the Azure Portal. Within about 24 hours of your cluster being up and running, you should start to see some alerts show up (Low and Medium).
```
Comment (ckittel) [inline docs/deploy/13-validation-logs.md:198]:
```suggestion
Microsoft Defender for Containers reviews your cluster's logs and detects runtime workload behavior that might be undesired. [Those alerts](https://docs.microsoft.com/azure/defender-for-cloud/alerts-reference#alerts-k8scluster) surface in the Azure Resource Graph (`securityresources` -> `microsoft.security/locations/alerts`) and also in Microsoft Defender for Cloud in the Azure Portal. Within about 24 hours of your cluster being up and running, you may start to see some Low and Medium alerts show up.
```

**Review by ckittel — CHANGES_REQUESTED** [review]
Looks really good @magrande!  I've got a few suggestions here for you to review.

File: docs/deploy/01-prerequisites.md (hunk 1 above)
Comment (magrande) [reply docs/deploy/01-prerequisites.md]: Right, no longer needed. Removed

File: docs/deploy/04-subscription.md (hunk 4 above)
Comment (magrande) [reply docs/deploy/04-subscription.md]: Done !

File: docs/deploy/04-subscription.md (hunk 2 above)
Comment (magrande) [reply docs/deploy/04-subscription.md]: Done, but should we then use that exact name in subscription.json, line 251 ?

"displayName": "Microsoft Defender for Containers should be enabled in the cluster.", ----->
                "displayName": "No AKS clusters without Microsoft Defender for Containers.",

File: docs/deploy/01-prerequisites.md [hunk 10]
```diff
@@ -39,6 +39,7 @@ Throughout this walkthrough, take note of the following symbol.
    # Keep running until all say "Registered." (This may take up to 20 minutes.)
    az feature list -o table --query "[?name=='Microsoft.ContainerService/EnablePodIdentityPreview'].{Name:name,State:properties.state}"
 
+
```
Comment (ckittel) [inline docs/deploy/01-prerequisites.md:42]: Left a trailing a space after your undo.

```suggestion
```

File: docs/deploy/04-subscription.md (hunk 2 above)
Comment (ckittel) [reply docs/deploy/04-subscription.md]: How "off" are the others in this list?  Do we need to just align them closer for all?

File: docs/deploy/04-subscription.md [hunk 11]
```diff
@@ -55,11 +56,10 @@ Also, depending on your workload subscription scope, some of the policies applie
...
 As mentioned in the Azure Policy section above, we enable the following Microsoft Defender for Cloud's services.
 
-* [Microsoft Defender for Kubernetes](https://docs.microsoft.com/azure/security-center/defender-for-kubernetes-introduction)
-* [Microsoft Defender for Container Registries](https://docs.microsoft.com/azure/security-center/defender-for-container-registries-introduction)
-* [Microsoft Defender for Key Vault](https://docs.microsoft.com/azure/security-center/defender-for-key-vault-introduction)
-* [Microsoft Defender for Azure DNS](https://docs.microsoft.com/azure/security-center/defender-for-dns-introduction)
-* [Microsoft Defender for Azure Resource Manager](https://docs.microsoft.com/azure/security-center/defender-for-resource-manager-introduction)
+* [Microsoft Defender for Containers](https://docs.microsoft.com/azure/defender-for-cloud/defender-for-containers-introduction)
+* [Microsoft Defender for Key Vault](https://docs.microsoft.com/azure/defender-for-cloud/defender-for-key-vault-introduction)
+* [Microsoft Defender for Azure DNS](hhttps://docs.microsoft.com/azure/defender-for-cloud/defender-for-dns-introduction)
```
Comment (ckittel) [inline docs/deploy/04-subscription.md:61]:
```suggestion
* [Microsoft Defender for Azure DNS](https://docs.microsoft.com/azure/defender-for-cloud/defender-for-dns-introduction)
```

File: docs/deploy/01-prerequisites.md (hunk 1 above)
Comment (ckittel) [reply docs/deploy/01-prerequisites.md]: Thanks!

File: docs/deploy/04-subscription.md (hunk 4 above)
Comment (ckittel) [reply docs/deploy/04-subscription.md]: Thanks

File: docs/deploy/04-subscription.md (hunk 2 above)
Comment (ckittel) [reply docs/deploy/04-subscription.md]: It's okay if they are "off" for now.  Looks like we're pretty consistent with their presentation here.  So we'll just run with it as is.

File: docs/deploy/04-subscription.md [hunk 12]
```diff
@@ -40,6 +40,8 @@ To help govern our resources, there are policies we apply over the scope of thes
 | No AKS clusters without Azure Policy | rg-bu0001a0005            | Restricts the creation of AKS clusters to only those that have the Azure Policy Add-on enabled.   |
 | No AKS clusters without BYOK OS & Data Disk Encryption | rg-bu0001a0005  | Restricts the creation of AKS clusters to only those that have customer-managed disk encryption enabled. (_This is in audit only mode, as not all customers may wish to do this._) |
 | No AKS clusters without encryption-at-host | rg-bu0001a0005      | Restricts the creation of AKS clusters to only those that have the Encryption-At-Host feature enabled. (_This is in audit only mode, as not all customers may wish to do this._) |
+| No AKS clusters without Microsoft Defender for Containers | rg-bu0001a0005                | Restricts the creation of AKS clusters to only those that have the Microsoft Defender for Containers feature enabled. |
+
```
Comment (ckittel) [inline docs/deploy/04-subscription.md:44]: Sorry, I left a dangling newline there. Fixing.

```suggestion
```

**Review by ckittel — APPROVED** [review]
LGTM!

**Pull request merged**